packages/rest/README.md

```diff
@@ -0,0 +1,94 @@
+# @loopback.rest
+
+The REST API package for [loopback-next](https://github.com/strongloop/loopback-next).
+
+## STATUS: ALPHA
+This package is not yet ready for production use, and the API has not yet
+stabilized. Additionally, this document is also a work-in-progress and some
+sections may not have corresponding code!
+
+## Overview
+This component provides a REST server for your application instances, complete
+with:
+- new custom routing engine (special thanks to @bajtos)!
+- tools for defining your application routes
+- OpenAPI 2.0 spec (`swagger.json`/`swagger.yaml`) generation
+- a default sequence implementation to manage the request and response lifecycle
+## Installation
+To use this package, you'll need to install both `@loopback/core` and
+`@loopback/rest`.
+
+```shell
+$ npm install --save @loopback/core @loopback/rest
+```
+
+## Basic Use
+Here's a basic "Hello World" application using `@loopback/core` and
+`@loopback/rest`:
+
+  ```ts
+  import {Application} from '@loopback/core';
+  import {RestComponent, RestServer} from '@loopback/rest';
+
+  const app = new Application({
+    components: [
+      RestComponent,
+    ]
+  });
+  const rest = await app.getServer(RestServer);
+  rest.handler((sequence, request, response) => {
+    sequence.send(response, 'hello world');
+  });
+
+  (async function start() {
+    await app.start();
+    console.log(`REST server running on port: ${rest.getSync('http.port')}`);
+  })();
+  ```
+
+## Configuration
+The rest package is configured by passing a `rest` property inside of your
+Application options.
+
+```ts
+  const app = new Application({
+    components: [
+      RestComponent,
+    ],
+    rest: {
+      port: 3001
+    }
+  });
+```
```

Comment:
For the reference, maybe a better way is to move `rest` object to be next to RestComponent to avoid one extra level of redirection, for example:
```ts
const app = new Application({
  components: [
    {type: RestComponent, config: {port: 3001}}
  ],
```

Reply:
The advantage of this extra level of indirection is that with dynamic bindings, it's possible to decouple the code telling the app which components to use from the code deciding how to configure these components, and allow applications to dynamically fetch the configuration (e.g. from https://www.consul.io/), or perhaps even re-configure them the fly.

Reply:
OTOH, having the configuration for components address the top-level properties means that component developers might end up competing for keys on the config object.

Reply:
To allow that, the components should become an object with keys to represent names for component instances. For example:
```js
components: {
  FooCorpWidgetScape, 
  FooCorpSamoflangeWizard1: FooCorpSamoflangeWizard,
  FooCorpSamoflangeWizard2: FooCorpSamoflangeWizard,
}
component-config: {
  FooCorpWidgetScape: {...},
  FooCorpSamoflangeWizard1: {x: 1},
  FooCorpSamoflangeWizard2: {x: 2},
}
```

Reply:
I agree with both proposals.

As far as `@loopback/core` is concerned, I think we can keep the naming convention open, at least for the Beta release.

The only change I see as necessary (because it's a breaking change) is to allow named component registration - i.e. change `components: [FooCorpWidgetScape]` to `components: {FooCorpWidgetScape}`.

I think this should be done in a standalone pull request though, outside of scope of the work on extracting the REST component from Core. This pull request is already taking too long to finish.